lib/web_ui/lib/src/engine/clipboard.dart

```diff
@@ -91,9 +91,49 @@ class ClipboardAPIPasteStrategy implements PasteFromClipboardStrategy {
 class ExecCommandCopyStrategy implements CopyToClipboardStrategy {
   @override
   void setData(String text) {
-    // TODO(nurhan): https://github.com/flutter/flutter/issues/48578
-    print('Clipboard is only implemented for browsers '
-        'supporting Clipboard API. Use context menu for text editing.');
+    final html.TextAreaElement tempTextArea = _appendTemporaryTextArea();
+    tempTextArea.text = text;
+
+    tempTextArea.focus();
+
+    final html.Range range = html.document.createRange();
+    range.selectNodeContents(tempTextArea);
+    html.window.getSelection().removeAllRanges();
+    html.window.getSelection().addRange(range);
```

Comment:
In `text_editing.dart` we only do `element.setSelectionRange(...)`. Is there a reason you had to use the `Range` API here? I think the `Range` API is used for `content-editable` elements, and we are not using that, we are using a `<textarea>`.

Reply:
Yes, the code we are using is from here: https://developers.google.com/web/updates/2015/04/cut-and-copy-commands

Reply:
Thanks for sharing the link. I skimmed through it. It seems like they are using the `Range` API in the first example. But in the 2nd example (textarea):
```javascript
var cutTextareaBtn = document.querySelector('.js-textareacutbtn');

cutTextareaBtn.addEventListener('click', function(event) {
  var cutTextarea = document.querySelector('.js-cuttextarea');
  cutTextarea.select();

  try {
    var successful = document.execCommand('cut');
    var msg = successful ? 'successful' : 'unsuccessful';
    console.log('Cutting text command was ' + msg);
  } catch(err) {
    console.log('Oops, unable to cut');
  }
});
```

Reply:
I changed to using textArea.value then textArea.select(). Looks like working in Safari :)